src/traces/scattergl/index.js

```diff
@@ -13,10 +13,12 @@ var ScatterGl = {};
 ScatterGl.attributes = require('./attributes');
 ScatterGl.supplyDefaults = require('./defaults');
 ScatterGl.colorbar = require('../scatter/colorbar');
+ScatterGl.hoverPoints = require('../scatter/hover');
 
 // reuse the Scatter3D 'dummy' calc step so that legends know what to do
-ScatterGl.calc = require('../scatter3d/calc');
+ScatterGl.calc = require('../scatter/calc');
```

Comment:
With 6ac722d4d440bbff10568cecc46174076071f253 we're back in the ~1500ms for 1e6 first render as on `master` :tada: 

Most of `scattergl/calc.js` is now skipped when dragmode is set to pan/zoom (the default).

 Calling `Plotly.relayout` from pan/zoom to lasso/select now trigger a recalc where `gd.calcdata` is filled so that the selection logic knows what to do.